pkg/logs/input/container/docker.go

```diff
@@ -208,6 +227,35 @@ func (dt *DockerTailer) forwardMessages() {
 	}
 }
 
+// fetchLogsMetadata returns the logs metedata defined in container labels.
+// if no metadata are specified, return false
+func (dt *DockerTailer) fetchLogsMetadata() (*config.LogsConfig, bool) {
+	information, err := dt.cli.ContainerInspect(context.Background(), dt.ContainerID)
+	if err != nil {
+		return nil, false
+	}
+	return dt.extractLogsMetadataFromLabels(information.Config.Labels)
+}
+
+// extractLogsMetadataFromLabels returns logs metadata extracted from labels when it contains key 'com.datadoghq.ad.logs'
+// and its value respects format '[{...}]'.
+func (dt *DockerTailer) extractLogsMetadataFromLabels(labels map[string]string) (*config.LogsConfig, bool) {
+	label, exists := labels[logsMetadataPath]
+	if !exists {
+		return nil, false
+	}
+	var configs []config.LogsConfig
+	err := json.Unmarshal([]byte(label), &configs)
```

Comment:
Another reason why you don't want a `LogsConfig`, you don't want parsing to fail for unrelated stuff.